src/styles/main.scss

```diff
@@ -0,0 +1,70 @@
+$cell-size: 100px;
+$gap-size: 1px;
+$days-in-week: 7;
+$days: (
+  'mon': 0,
+  'tue': 1,
+  'wed': 2,
+  'thu': 3,
+  'fri': 4,
+  'sat': 5,
+  'sun': 6,
+);
+
+body {
+  margin: 0;
+  height: 100vh;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap-size;
+  width: $days-in-week * ($cell-size + $gap-size);
+  padding: 20px;
+
+  &__day {
+    box-sizing: border-box;
+    height: $cell-size;
+    width: $cell-size;
+    position: relative;
+    background-color: #eee;
+    border: 1px solid black;
+    transition: all 0.5s;
+
+    @for $i from 1 through 31 {
+      &:nth-child(#{$i})::before {
+        content: '#{$i}';
+        font-family: Arial, sans-serif;
+        font-size: 30px;
+        line-height: 30px;
+        position: absolute;
+        text-align: center;
+        bottom: ($cell-size - 2 * $gap-size) / 2;
+        right: ($cell-size - 3 * $gap-size) / 2;
+        transform: translate(50%, 50%);
```

Comment:
- These properties are the same for each element, so move them outside the loop
- It's better to avoid position absolute, in a case when you need to center the element you can use display flex

Reply:
As for text centering in ::before, unfortunately, my tests crashed when using flex, so I had to reinvent the wheel this way...
In general, if it is possible to pass the task by bypassing the tests, then absolute positioning and the bottom and right tasks can be removed (they only position the text)